localboot/grub.go

```diff
@@ -95,35 +124,53 @@ func ParseGrubCfg(grubcfg string, basedir string, grubVersion int) []bootconfig.
 	return bootconfigs
 }
 
+func unquote(ver grubVersion, text string) string {
+	if ver == grubV2 {
+		// if grub2, unquote the string, as directives could be quoted
+		// https://www.gnu.org/software/grub/manual/grub/grub.html#Quoting
+		// TODO unquote everything, not just \$
+		return strings.Replace(text, `\$`, "$", -1)
+	}
+	// otherwise return the unmodified string
+	return text
+}
+
 // ScanGrubConfigs looks for grub2 and grub legacy config files in the known
 // locations and returns a list of boot configurations.
 func ScanGrubConfigs(basedir string) []bootconfig.BootConfig {
 	bootconfigs := make([]bootconfig.BootConfig, 0)
-	// Scan Grub 2 configurations
-	for _, grubpath := range Grub2Paths {
-		path := path.Join(basedir, grubpath)
-		log.Printf("Trying to read %s", path)
-		grubcfg, err := ioutil.ReadFile(path)
-		if err != nil {
-			log.Printf("cannot open %s: %v", path, err)
-			continue
+	err := filepath.Walk(basedir, func(currentPath string, info os.FileInfo, err error) error {
+		if path.Dir(currentPath) == basedir && info.IsDir() && !isGrubSearchDir(path.Base(currentPath)) {
```

Comment:
`currentPath` should be normalized, and `err` should be checked as first thing

Reply:
done